my-app/src/BtnNew.tsx

```diff
@@ -0,0 +1,11 @@
+import { Link } from 'react-router-dom';
+
+export function BtnNew() {
+  return (
+    <Link to="/form">
+      <button type="button" className="new-entry-button" data-view="entry-form">
```

Comment:
Did you use `<button>` just for styling purposes? With `Link` there is no need to wrap a button. You can just style it directly and put in the text:
`<Link to="/form" className="new-entry-button">New</Link>`

Also, I'm not sure why you use `data-view`? That attribute is not needed with React.